core/utils/fork-tree/src/lib.rs

```diff
@@ -1203,18 +1184,36 @@ mod test {
 
 		tree.prune(
 			&"C",
-			3,
+			&3,
 			&is_descendent_of,
+			&|_| true,
+		).unwrap();
+
+		assert_eq!(
+			tree.roots.iter().map(|node| node.hash).collect::<Vec<_>>(),
+			vec!["B"],
+		);
+
+		assert_eq!(
+			tree.iter().map(|(hash, _, _)| *hash).collect::<Vec<_>>(),
+			vec!["B", "C", "D", "E"],
+		);
+
+		tree.prune(
+			&"E",
+			&5,
+			&is_descendent_of,
+			&|_| true,
 		).unwrap();
 
 		assert_eq!(
 			tree.roots.iter().map(|node| node.hash).collect::<Vec<_>>(),
-			vec!["C"],
+			vec!["D"],
 		);
 
 		assert_eq!(
 			tree.iter().map(|(hash, _, _)| *hash).collect::<Vec<_>>(),
-			vec!["C", "D", "E"],
+			vec!["D", "E"],
```

Comment:
`"D"` is an ancestor of `"E"`.  Why is it kept?

Reply:
It doesn't prune the parent.